awswrangler/s3/_write_parquet.py

```diff
@@ -131,6 +158,41 @@ def _to_parquet_chunked(
     return proxy.close()  # blocking
 
 
+def _to_parquet_distributed(  # pylint: disable=unused-argument
+    df: pd.DataFrame,
+    schema: pa.Schema,
+    index: bool,
+    compression: Optional[str],
+    compression_ext: str,
+    pyarrow_additional_kwargs: Optional[Dict[str, Any]],
+    cpus: int,
+    dtype: Dict[str, str],
+    boto3_session: Optional[boto3.Session],
+    s3_additional_kwargs: Optional[Dict[str, str]],
+    use_threads: Union[bool, int],
+    path: Optional[str] = None,
+    path_root: Optional[str] = None,
+    filename_prefix: Optional[str] = "",
+    max_rows_by_file: Optional[int] = 0,
+    bucketing: bool = False,
+) -> List[str]:
+    if bucketing:
+        # Add bucket id
+        filename_prefix = f"{filename_prefix}_bucket-{df.name:05d}"
+    # Create Ray Dataset
+    ds = from_modin(df) if isinstance(df, ModinDataFrame) else from_pandas(df)
+    # Repartition by max_rows_by_file
+    if max_rows_by_file and (max_rows_by_file > 0):
+        ds = ds.repartition(math.ceil(ds.count() / max_rows_by_file))
+    datasource = ParquetDatasource()
+    ds.write_datasource(
+        datasource,  # type: ignore
```

Comment:
So no more CustomerBlockWritePathProvider?

Reply:
No, it was more of a workaround. Found the way to achieve the same with default block path provider.

One thing it doesn't do yet is in case of bucketing, it adds both bucket id and block id to the prefix. Happy to discuss this too

Reply:
Updated bucketing keys